packages/analytics-browser/src/browser-client.ts

```diff
@@ -193,17 +160,17 @@ export class AmplitudeBrowser extends AmplitudeCore implements BrowserClient {
           session_id: previousSessionId,
           time: previousLastEventTime + 1,
         };
-        eventOptions.device_id = this.previousSessionDeviceId;
-        eventOptions.user_id = this.previousSessionUserId;
+        eventOptions.device_id = this.config.lastSessionDeviceId;
+        eventOptions.user_id = this.config.lastSessionUserId;
         this.track(DEFAULT_SESSION_END_EVENT, undefined, eventOptions);
       }
 
       this.track(DEFAULT_SESSION_START_EVENT, undefined, {
         session_id: sessionId,
         time: sessionId - 1,
       });
-      this.previousSessionDeviceId = this.config.deviceId;
-      this.previousSessionUserId = this.config.userId;
+      this.config.lastSessionDeviceId = this.config.deviceId;
```

Comment:
super nit: I think `previousSessionDeviceId` is a slightly better name as `last` can also imply "final". Just my 2c's.

Reply:
Updated to `previousSessionDeviceId` and `previousSessionUserId`